llmfoundry/optim/scheduler.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import textwrap
+import warnings
+from typing import Union
+
+from composer.core import State, Time, TimeUnit
+from composer.optim import ComposerScheduler, LinearScheduler
+from composer.optim.scheduler import _convert_time
+
+__all__ = ['InverseSquareRootWithWarmupScheduler']
+
+
+def _raise_if_units_dont_match(time: Union[str, Time],
+                               t_max: Union[str, Time]) -> None:
+    if isinstance(time, str):
+        time = Time.from_timestring(time)
+    if isinstance(t_max, str):
+        t_max = Time.from_timestring(t_max)
+    if time.unit != t_max.unit:
+        raise ValueError(
+            'All time units must be the same as max_duration units.')
+
+
+def _raise_if_units_dur(time: Union[str, Time]) -> None:
+    if isinstance(time, str):
+        time = Time.from_timestring(time)
+    if time.unit == TimeUnit('dur'):
+        raise ValueError(
+            't_warmup, t_scale, and t_cooldown cannot be in units of "dur".')
```

Comment:
Raise the error in a way that the variable name and its value are printed

Reply:
Done